### PR TITLE
Blackened code

### PR DESCRIPTION
**Pull request by jace**
With one special fix for Flask-FlatPages and one bugfix in `models/helpers.py` where some reserved keywords were missing commas.

### CODE REVIEW

File: funnel/extapi/boxoffice.py [hunk 1]
```diff
@@ -31,18 +36,24 @@ def get_tickets(self, ic):
...
+                            'twitter': extract_twitter_handle(
+                                line_item.get('assignee').get('twitter', '')
+                            ),
+                            'company': line_item.get('assignee').get('company'),
+                            'city': line_item.get('assignee').get('city', ''),
+                            'job_title': line_item.get('assignee').get('jobtitle', ''),
+                            'ticket_no': unicode(line_item.get('line_item_seq')),
+                            'ticket_type': line_item.get('item', {}).get('title', '')[
+                                :80
+                            ],
```
Comment (iambibhas) [inline funnel/extapi/boxoffice.py]: This one is a bit ridiculous. Maybe ignore it.

File: funnel/forms/profile.py [hunk 2]
```diff
@@ -1,29 +1,52 @@
...
 class NewProfileForm(forms.Form):
     """Create a new profile."""
 
-    profile = forms.RadioField(__("Organization"), validators=[forms.validators.DataRequired("Select an organization")],
-        description=__(u"Select the organization you’d like to create a Talkfunnel for"))
+    profile = forms.RadioField(
+        __("Organization"),
+        validators=[forms.validators.DataRequired("Select an organization")],
+        description=__(
+            u"Select the organization you’d like to create a Talkfunnel for"
```
Comment (iambibhas) [inline funnel/forms/profile.py]: Need to go through the repo and remove all references to Talkfunnel/funnel.

File: funnel/forms/profile.py [hunk 3]
```diff
@@ -1,29 +1,52 @@
...
+    )
+    admin_team = QuerySelectField(
+        u"Admin Team",
         validators=[forms.validators.DataRequired(_(u"Please select a team"))],
-        get_label='title', allow_blank=False,
-        description=__("The team of users with administrative rights to this Talkfunnel (owners always have admin access)"))
+        get_label='title',
+        allow_blank=False,
+        description=__(
+            "The team of users with administrative rights to this Talkfunnel (owners always have admin access)"
```
Comment (iambibhas) [inline funnel/forms/profile.py]: Anything can be done for looong strings? Break them to multiple lines?

File: funnel/forms/project.py [hunk 4]
```diff
@@ -5,81 +5,147 @@
...
+    'ProjectForm',
+    'CfpForm',
+    'ProjectTransitionForm',
+    'RsvpForm',
+    'TicketClientForm',
+    'TicketTypeForm',
+    'ProjectBoxofficeForm',
+    'SavedProjectForm',
+    'ProjectScheduleTransitionForm',
+    'ProjectCfpTransitionForm',
```
Comment (iambibhas) [inline funnel/forms/project.py]: This list doesn't seem sorted. Did we remove sorting of imports?

File: funnel/forms/proposal.py [hunk 5]
```diff
@@ -2,37 +2,57 @@
...
-    return ProposalLabelForm(obj=proposal.formlabels if proposal else None, meta={'csrf': False})
+            setattr(
+                ProposalLabelForm,
+                label.name,
+                forms.RadioField(
+                    label.form_label_text,
+                    description=label.description,
+                    validators=[forms.validators.DataRequired(__("Please select one"))]
+                    if label.required
+                    else [],
```
Comment (iambibhas) [inline funnel/forms/proposal.py]: Any option to indent these?

File: funnel/forms/proposal.py [hunk 6]
```diff
@@ -41,87 +61,163 @@ class ProposalLabelAdminForm(forms.Form):
...
-                ))
+            setattr(
+                ProposalLabelAdminForm,
+                label.name,
+                field_type(
+                    label.form_label_text,
+                    description=label.description,
+                    validators=[forms.validators.DataRequired(__("Please select one"))]
+                    if label.required
+                    else [],
```
Comment (iambibhas) [inline funnel/forms/proposal.py]: Same as above. can we indent these?

File: funnel/forms/proposal.py [hunk 7]
```diff
@@ -41,87 +61,163 @@ class ProposalLabelAdminForm(forms.Form):
...
+    )
+    preview_video = forms.URLField(
+        __("Preview Video"),
+        validators=[
+            forms.validators.Optional(),
+            forms.validators.URL(),
+            forms.validators.ValidUrl(),
+        ],
+        description=__(
+            "Link to your preview video. Use a video to engage the community and give them a better idea about what you are planning to cover in your session and why they should attend. "
```
Comment (iambibhas) [inline funnel/forms/proposal.py]: Very long string here.

File: funnel/jobs/jobs.py [hunk 8]
```diff
@@ -42,17 +48,28 @@ def tag_locations(project_id):
                 for item in results:
                     geoname = item.get('geoname', {})
                     if geoname:
-                        geonames[geoname['geonameid']]['geonameid'] = geoname['geonameid']
-                        geonames[geoname['geonameid']]['primary'] = geonames[geoname['geonameid']].get('primary', True)
+                        geonames[geoname['geonameid']]['geonameid'] = geoname[
+                            'geonameid'
+                        ]
```
Comment (iambibhas) [inline funnel/jobs/jobs.py]: Breaking dict lookup like this is a bit unusual.

File: funnel/models/commentvote.py [hunk 9]
```diff
@@ -13,13 +13,18 @@
...
     PUBLIC = (0, 'public', __("Public"))
     SCREENED = (1, 'screened', __("Screened"))
     HIDDEN = (2, 'hidden', __("Hidden"))
     SPAM = (3, 'spam', __("Spam"))
-    DELETED = (4, 'deleted', __("Deleted"))  # For when there are children to be preserved
+    DELETED = (
+        4,
+        'deleted',
+        __("Deleted"),
+    )  # For when there are children to be preserved
```
Comment (iambibhas) [inline funnel/models/commentvote.py]: If we move the comment to above or below this line, it'll restore the items like the ones above.

File: funnel/models/contact_exchange.py [hunk 10]
```diff
@@ -163,9 +195,13 @@ def contacts_for_project_and_date(cls, user, project, date, archived=False):
...
             Participant.project_id == project.id,
             db.cast(
-                db.func.date_trunc('day', db.func.timezone(project.timezone.zone, cls.scanned_at)),
-                db.Date) == date
+                db.func.date_trunc(
+                    'day', db.func.timezone(project.timezone.zone, cls.scanned_at)
+                ),
+                db.Date,
             )
+            == date,
```
Comment (iambibhas) [inline funnel/models/contact_exchange.py]: Is this fine?

File: funnel/models/event.py [hunk 11]
```diff
@@ -168,13 +222,33 @@ def checkin_list(cls, event):
...
             FROM participant INNER JOIN attendee ON participant.id = attendee.participant_id LEFT OUTER JOIN sync_ticket ON participant.id = sync_ticket.participant_id
             WHERE attendee.event_id = :event_id
             ORDER BY participant.fullname
-        ''')).params(event_id=event.id).all()
+        '''
+                )
+            )
+            .params(event_id=event.id)
+            .all()
+        )
```
Comment (iambibhas) [inline funnel/models/event.py]: This part seems a bit messed up. Should the closing `'''` be indented a bit?

File: funnel/models/helpers.py [hunk 12]
```diff
@@ -30,8 +30,8 @@
     'contacts',
     'delete',
     'edit',
-    'email'
-    'emails'
+    'email',
+    'emails',
```
Comment (iambibhas) [inline funnel/models/helpers.py]: So this was broken till now.. -_-

File: funnel/models/project.py [hunk 13]
```diff
@@ -246,103 +310,165 @@ def datelocation(self):
...
         return u', '.join(filter(None, [daterange, self.location]))
 
-    schedule_state.add_conditional_state('PAST', schedule_state.PUBLISHED,
-        lambda project: project.schedule_end_at is not None and utcnow() >= project.schedule_end_at,
+    schedule_state.add_conditional_state(
+        'PAST',
+        schedule_state.PUBLISHED,
+        lambda project: project.schedule_end_at is not None
+        and utcnow() >= project.schedule_end_at,
         lambda project: db.func.utcnow() >= project.schedule_end_at,
```
Comment (iambibhas) [inline funnel/models/project.py]: having `and` from previous statement and this statement in same indent level is a bit confusing.

File: funnel/models/project.py [hunk 14]
```diff
@@ -367,20 +493,29 @@ def _validate_name(self, key, value):
...
+                db.text(
+                    '''
             SELECT DATE_TRUNC('day', "start_at" AT TIME ZONE :timezone) AS date, COUNT(*) AS count
             FROM "session" WHERE "project_id" = :project_id AND "start_at" IS NOT NULL AND "end_at" IS NOT NULL
             GROUP BY date ORDER BY date;
-            ''')).params(timezone=self.timezone.zone, project_id=self.id))
+            '''
+                )
+            )
+            .params(timezone=self.timezone.zone, project_id=self.id)
```
Comment (iambibhas) [inline funnel/models/project.py]: Same concern about the `'''` here.

File: funnel/views/commentvote.py [hunk 15]
```diff
@@ -88,40 +90,84 @@ def new_comment(self):
...
+                    ).first_or_404()
+                    if (
+                        parent.user.email
+                    ):  # FIXME: https://github.com/hasgeek/funnel/pull/324#discussion_r241270403
+                        if (
+                            parent.user == self.obj.user
+                        ):  # check if parent comment & proposal owner are same
+                            if (
+                                not current_auth.user == parent.user
+                            ):  # check if parent comment is by proposal owner
```
Comment (iambibhas) [inline funnel/views/commentvote.py]: If we move the comments below the condition check, then we can have single lines for them.

Comment (iambibhas) [reply funnel/views/commentvote.py]: Actually, leave this. I'll fix it in the comment PR.

File: funnel/views/project.py [hunk 16]
```diff
@@ -55,17 +55,27 @@ def project_data(project):
...
+        'start_at': project.schedule_start_at.astimezone(project.timezone)
+        .date()
+        .isoformat()
+        if project.schedule_start_at
+        else None,
+        'end_at': project.schedule_end_at.astimezone(project.timezone)
+        .date()
+        .isoformat()
+        if project.schedule_end_at
+        else None,
```
Comment (iambibhas) [inline funnel/views/project.py]: These 2 seems a bit messed up.

File: funnel/views/schedule.py [hunk 17]
```diff
@@ -73,32 +79,37 @@ def schedule_data(project):
...
+                'json_url': session.proposal.url_for('json', _external=True)
+                if session.proposal
+                else None,
+                'proposal_url': session.proposal.url_for(_external=True)
+                if session.proposal
+                else None,
+                'proposal': session.proposal.suuid if session.proposal else None,
+                'feedback_url': session.url_for('feedback', _external=True)
+                if session.proposal
+                else None,
```
Comment (iambibhas) [inline funnel/views/schedule.py]: Same with these.

File: funnel/views/schedule.py [hunk 18]
```diff
@@ -185,15 +201,22 @@ def subscribe_schedule(self):
...
+        )
 
     @route('ical')
     @requires_permission('view')
     def schedule_ical(self):
         cal = Calendar()
-        cal.add('prodid', "-//Schedule for {event}//funnel.hasgeek.com//".format(event=self.obj.title))
+        cal.add(
+            'prodid',
+            "-//Schedule for {event}//funnel.hasgeek.com//".format(
```
Comment (iambibhas) [inline funnel/views/schedule.py]: this url needs to be updated.

File: funnel/views/schedule.py [hunk 19]
```diff
@@ -266,25 +301,34 @@ class ScheduleVenueRoomView(VenueRoomViewMixin, UrlForView, ModelView):
...
     def schedule_room_ical(self):
         cal = Calendar()
-        cal.add('prodid', "-//Schedule for room {room} at {venue} for {event}//funnel.hasgeek.com//".format(
-            room=self.obj.title,
-            venue=self.obj.venue.title,
-            event=self.obj.venue.project.title,
-        ))
+        cal.add(
+            'prodid',
+            "-//Schedule for room {room} at {venue} for {event}//funnel.hasgeek.com//".format(
```
Comment (iambibhas) [inline funnel/views/schedule.py]: Same here.

File: funnel/views/session.py [hunk 20]
```diff
@@ -109,11 +120,19 @@ def view(self):
...
-            'sessions': session_list_data(self.obj.project.scheduled_sessions, with_modal_url='view_popup'),
+            'sessions': session_list_data(
+                self.obj.project.scheduled_sessions, with_modal_url='view_popup'
+            ),
             # FIXME: This timezone by UTC offset is not accounting for DST. Look up where it's being used and fix it
-            'timezone': utcnow().astimezone(self.obj.project.timezone).utcoffset().total_seconds(),
+            'timezone': utcnow()
+            .astimezone(self.obj.project.timezone)
+            .utcoffset()
+            .total_seconds(),
```
Comment (iambibhas) [inline funnel/views/session.py]: The following lines could be indented for better readability.

File: funnel/forms/proposal.py (hunk 5 above)
Comment (jace) [reply funnel/forms/proposal.py]: Black doesn't have options.

File: funnel/forms/profile.py (hunk 3 above)
Comment (jace) [reply funnel/forms/profile.py]: Yes.

File: funnel/forms/project.py (hunk 4 above)
Comment (jace) [reply funnel/forms/project.py]: isort doesn't appear to touch `__all__`

File: funnel/views/schedule.py (hunk 18 above)
Comment (jace) [reply funnel/views/schedule.py]: We're not doing non-formatting related fixes in this PR.

File: funnel/extapi/boxoffice.py (hunk 1 above)
Comment (jace) [reply funnel/extapi/boxoffice.py]: That's because it hit the 88 column limit.

File: funnel/models/contact_exchange.py (hunk 10 above)
Comment (jace) [reply funnel/models/contact_exchange.py]: Yes. Binary operators can be on a line of their own. Black compares bytecode of pre- and post-format to confirm there are no syntactical changes.

File: funnel/models/event.py (hunk 11 above)
Comment (jace) [reply funnel/models/event.py]: Fixed

File: funnel/views/project.py (hunk 16 above)
Comment (jace) [reply funnel/views/project.py]: Fixed using parentheses.

**Pull request merged**